src/Libraries/RevitNodes/Elements/FamilyInstance.cs

```diff
@@ -184,6 +276,29 @@ private void InternalSetPosition(XYZ fi)
             TransactionManager.Instance.TransactionTaskDone();
         }
 
+        private void InternalSetPosition(Line pos)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+
+            //var lp = InternalFamilyInstance.get_Parameter(BuiltInParameter.CURVE_IS_REFERENCE_LINE).Set(pos.Reference.ElementId);
+            var lp = InternalFamilyInstance.Location as LocationCurve;
+            lp.Curve = pos;
+
+            TransactionManager.Instance.TransactionTaskDone();
+        }
+
+        private void InternalSetFace(Reference face)
+        {
+            if (InternalFamilyInstance.HostFace == face)
+                return;
+
+            TransactionManager.Instance.EnsureInTransaction(Document);
+
+            InternalFamilyInstance.get_Parameter(BuiltInParameter.FAMILY_WORK_PLANE_BASED).Set(face.ElementId);
+
+            TransactionManager.Instance.TransactionTaskDone();
```

Comment:
Looks like FAMILY_WORK_PLANE_BASED parameter accepts bool value. Have you tested it? https://knowledge.autodesk.com/search-result/caas/CloudHelp/cloudhelp/2015/ENU/Revit-API/files/GUID-DC143EB8-43CB-48AB-938E-7ADE3A9D2E63-htm.html